src/System.Private.CoreLib/src/System/Runtime/RuntimeImports.cs

```diff
@@ -507,17 +507,35 @@ internal enum GcRestrictedCalloutKind
         [RuntimeImport(RuntimeLibrary, "RhFindBlob")]
         internal static unsafe extern bool RhFindBlob(IntPtr hOsModule, uint blobId, byte** ppbBlob, uint* pcbBlob);
 
+#if CORERT
+        internal static uint RhGetLoadedModules(IntPtr[] resultArray)
+        {
+            // CORERT-TODO: This got broken within the last week or so. We don't care for single file.
+            if (resultArray != null)
```

Comment:
Yes, this got changed because of there is no 1:1 mapping between OS .dll and library for the multi file work.

To fix it:

- Change `RhGetModuleFromEEType` to `return *(this->m_ppModuleManager)` under CORERT ifdef.
- Change `FindBlob` to something like `return pModuleManager->GetModuleSection(300 + blobId)`. 
- Store pointer to metadata blobs via `ReadyToRunHeader` in `NodeFactory.cs`

Hopefully, these changes will make the whole thing much cleaner.